File: src/main/java/rag/retrieval/Hit.java
package rag.retrieval;

public record Hit(
        String chunkId,
        String docId,
        String source,
        String title,
        String text,
        double score
) {
    public Hit withScore(double newScore) {
        return new Hit(chunkId, docId, source, title, text, newScore);
    }

    @Override
    public String toString() {
        return chunkId + "@" + source + " score=" + score;
    }
}

File: src/main/python/rag/retrieval/hit.py
from dataclasses import dataclass, replace
from typing import Optional, List


@dataclass(frozen=True)
class Hit:
    doc_id: str
    source: str
    title: str
    text: str
    score: float
    embedding: Optional[List[float]] = None

    def with_score(self, new_score: float) -> 'Hit':
        return replace(self, score=new_score)

    def __str__(self) -> str:
        return f"{self.doc_id}@{self.source} score={self.score}"






